kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/WatchConnectionManager.java

```diff
@@ -64,11 +66,13 @@
   private final int reconnectInterval;
   private final AtomicInteger currentReconnectAttempt = new AtomicInteger(0);
   private final AtomicReference<WebSocket> webSocketRef = new AtomicReference<>();
-  private final ExecutorService executor = Executors.newSingleThreadExecutor();
+  private final ExecutorService reconnectExecutor = Executors.newSingleThreadExecutor();
```

Comment:
Why are we using two executors? A single ScheduledExecutorService thread pool with 2 threads would do. And most probably it could work with a single thread too, given that reconnect and ping are not supposed to happen in parallel (but that may be too much).